File: broker/src/main/scala/io/octopus/scala/factory/PublishThreadFactory.scala
package io.octopus.scala.factory


import io.netty.util.concurrent.FastThreadLocalThread

import java.util.concurrent.ThreadFactory
import java.util.concurrent.atomic.AtomicInteger

/**
 * @author chenxu
 * @version 1
 */

class PublishThreadFactory(poolName:String,threadGroup:ThreadGroup) extends ThreadFactory {

  private val nextId = new AtomicInteger(0)


  protected def newThread(r: Runnable, name: String) = new FastThreadLocalThread(this.threadGroup, r, name)

  override def newThread(r: Runnable): Thread = {
    val t = this.newThread(FastThreadLocalRunnable.wrap(r), this.poolName + "-" + this.nextId.incrementAndGet)
    t.setDaemon(false)
    t.setPriority(5)
    t
  }
}

File: broker-plugs/broker-mqtt/src/main/java/io/octopus/factory/PublishThreadFactory.java
//package io.octopus.factory;
//
//import io.netty.util.concurrent.FastThreadLocalThread;
//
//import java.util.concurrent.ThreadFactory;
//import java.util.concurrent.atomic.AtomicInteger;
//
//
///**
// * @author chenxu
// * @version 1
// */
//public class PublishThreadFactory implements ThreadFactory {
//
//    private final String poolName;
//    private final AtomicInteger nextId = new AtomicInteger(0);
//    private final ThreadGroup threadGroup;
//
//
//    public PublishThreadFactory(String poolName, ThreadGroup threadGroup) {
//        this.poolName = poolName;
//        this.threadGroup = threadGroup;
//    }
//
//
//    @Override
//    public Thread newThread(Runnable r) {
//        Thread t = this.newThread(FastThreadLocalRunnable.wrap(r), this.poolName +"-"+ this.nextId.incrementAndGet());
//        t.setDaemon(false);
//        t.setPriority(5);
//        return t;
//    }
//
//    protected Thread newThread(Runnable r, String name) {
//        return new FastThreadLocalThread(this.threadGroup, r, name);
//    }
//}
